From 5eded4476eb106b9f38233442e392322867cc253 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 27 Jun 2022 11:06:11 +0300
Subject: [PATCH 16/33] serial: linflex: Avoid devfdt_get_addr

Avoid the usage of devfdt_get_addr due to its incompatibility
with OF_LIVE.

Issue: ALB-9010
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/serial/serial_linflexuart.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/serial/serial_linflexuart.c b/drivers/serial/serial_linflexuart.c
index adadf7e839..389cede4f4 100644
--- a/drivers/serial/serial_linflexuart.c
+++ b/drivers/serial/serial_linflexuart.c
@@ -34,8 +34,6 @@
 #define LINFLEXD_UARTCR_ROSE		BIT(23)
 #define LINFLEX_LDIV_MULTIPLIER		(16)
 
-DECLARE_GLOBAL_DATA_PTR;
-
 struct linflex_fsl {
 	u32 lincr1;
 	u32 linier;
@@ -220,7 +218,7 @@ static int linflex_serial_probe(struct udevice *dev)
 	const char *clk_name = "lin";
 	int ret;
 
-	base_addr = devfdt_get_addr(dev);
+	base_addr = dev_read_addr(dev);
 	if (base_addr == FDT_ADDR_T_NONE)
 		return -EINVAL;
 
-- 
2.17.1

